### DISCUSSION TITLE
How to enforce JSON output using Function and Function call?

### DISCUSSION THREAD

**Discussion by ttdias25** [sashabaranov/go-openai#505] [Q&A]
I just read this [article](https://betterprogramming.pub/return-json-from-gpt-65d40bfc2ef6) , the examples are in javascript, i tried  to replicate it with this lib but go some errors, is there an example about how to use these 2 parameters? 

My code looks like this:

The jsonSchema variable is a plain text json schema string.
I didn't know what to pass to "FuncitonCall" since it can be "any"

```
	response, err := client.CreateChatCompletion(ctx, gpt.ChatCompletionRequest{
		Model:     model,
		Messages:  messages,
		MaxTokens: maxTokens,
		Functions: []gpt.FunctionDefinition{
			{
				Name:       "set_lessons",
				Parameters: jsonSchema,
			},
		},
		FunctionCall: "auto",
		Temperature: 0,
	})
```

The error returned was:

`{"type":"object","properties":{"title":{"type":"string"},"content":{"type":"string"}},"required":["title","content"]} is not of type 'object' - 'functions.0.parameters`

**Comment by tanakachitsamba** [comment]
With Json mode you wont need the function call to receive responses in JSON


```
response = client.chat.completions.create(
  model="gpt-3.5-turbo-1106",
  response_format={ "type": "json_object" },
  messages=[
    {"role": "system", "content": "You are a helpful assistant designed to output JSON."},
    {"role": "user", "content": "Who won the world series in 2020?"}
  ]
)
```
